src/Orleans.Transactions/State/TransactionalState.cs

```diff
@@ -207,7 +213,8 @@ Task ITransactionalResource.Abort(long transactionId)
         {
             if (!this.validState)
             {
-                await DoRecovery();
+                this.logger.Debug("Invalid state found. Recovering.");
```

Comment:
Maybe `IsEnabled` this

Reply:
We tend to do that when the string building is heavy due to complex arguments.  The logger should drop the message if it's disabled, so what we avoid with the pre check is ths cost of building the string, right?  or is there a cost here I'm not understanding.

Reply:
that's true, good point. Another way of looking at it is to always do the same thing for consistency, but I'm not firm on it either way.